flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/controller/FlinkResourceContext.java

```diff
@@ -47,6 +55,48 @@ public abstract class FlinkResourceContext<CR extends AbstractFlinkResource<?, ?
     private FlinkOperatorConfiguration operatorConfig;
     private Configuration observeConfig;
     private FlinkService flinkService;
+    private KubernetesJobAutoScalerContext autoScalerContext;
+
+    public KubernetesJobAutoScalerContext getJobAutoScalerContext() {
+        if (autoScalerContext != null) {
+            return autoScalerContext;
+        }
+        return autoScalerContext = createJobAutoScalerContext();
+    }
+
+    private KubernetesJobAutoScalerContext createJobAutoScalerContext() {
+        Configuration conf = new Configuration(getObserveConfig());
+        conf.set(
+                AutoScalerOptions.FLINK_CLIENT_TIMEOUT,
+                getOperatorConfig().getFlinkClientTimeout());
```

Comment:
Is this really necessary? seems costly, and we already have client timeout config in the operator itself which is already applied

Reply:
@mxm  also mentioned this, I have answered at this comment[1].

In brief, operator has client timeout config, however:
- `autoscaler` module cannot depend on operator code.
- The client timeout config of operator is used in many places(not only autoscaler), so move it into `autoscaler` module is strange.

[1] https://github.com/apache/flink-kubernetes-operator/pull/677#discussion_r1350028811

Reply:
Ah sorry, now I see. Let's keep this for now